tests/src/Interop/PInvoke/FixedBuffer/MarshalFixedBuffer.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Security;
+using TestLibrary;
+
+public unsafe class Managed
+{
+    private const int BufferSize = 10;
+
+    struct BlittableFixedBuffer
+    {
+        public fixed int elements[BufferSize];
+    }
+
+    struct NonBlittableFixedBuffer
+    {
+        public fixed bool elements[BufferSize];
+    }
+
+    [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
+    struct ElementAfterNonBlittableFixedBuffer
+    {
+        public fixed char fixedString[BufferSize];
```

Comment:
Is each character going to be marshalled to a single byte?

It is not the desired behavior wrt. globalization. It will mangle non-ASCII characters. The proper string marshaling should marshal the string as a whole using the right encoding.

Reply:
Yes currently it will marshal each character as a single byte. It treats it the same way that 10 consecutive `char` fields in a struct would be treated.

Reply:
Well, this means that this is cannot be used by anybody who builds production code with proper globalization. For example, we are not unlikely to be ever able to use it anywhere in our own framework.

Reply:
It works the same as if someone were to marshal a struct like below:

```
[StructLayout(LayoutKind.Sequential, CharSet=Ansi)]
struct ManualBuffer
{
    char c1;
    char c2;
    char c3;
    char c4;
    char c5;
    char c6;
    char c7;
    char c8;
    char c9;
    char c10;
}
```

So if the struct above is currently usable in production code with proper globalization, then this would be as well.

Reply:
One has to do manual marshaling for `fixed char` fields to get proper globalization today. And one will still have to do manual marshaling for them with this change. I do not think that this feature as implemented is helping in any real customer scenario (when it comes to strings at least).